pkg/graph/resolvers/model_plan_phase_test.go

```diff
@@ -0,0 +1,179 @@
+package resolvers
+
+import (
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/cmsgov/mint-app/pkg/storage"
+	"github.com/cmsgov/mint-app/pkg/userhelpers"
+
+	"github.com/cmsgov/mint-app/pkg/email"
+	"github.com/cmsgov/mint-app/pkg/graph/model"
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/shared/oddmail"
+)
+
+func (suite *ResolverSuite) TestSendEmailForPhaseSuggestionByModelPlanID() {
+	mockController := gomock.NewController(suite.T())
+	mockEmailService := oddmail.NewMockEmailService(mockController)
+	mockEmailTemplateService := email.NewMockTemplateService(mockController)
+
+	planName := "Plan For Milestones"
+	plan := suite.createModelPlan(planName)
+
+	collaboratorInput := &model.PlanCollaboratorCreateInput{
+		ModelPlanID: plan.ID,
+		UserName:    "CLAB",
+		TeamRoles:   []models.TeamRole{models.TeamRoleLeadership},
+	}
+
+	expectedEmail := "CLAB.doe@local.fake" // This comes from the stub fetch user info function
+
+	_, _, err := PlanCollaboratorCreate(
+		suite.testConfigs.Context,
+		suite.testConfigs.Store,
+		suite.testConfigs.Store,
+		suite.testConfigs.Logger,
+		nil,
+		nil,
+		email.AddressBook{},
+		collaboratorInput,
+		suite.testConfigs.Principal,
+		false,
+		userhelpers.GetUserInfoAccountInfoWrapperFunc(suite.stubFetchUserInfo),
+		false,
+	)
+	suite.NoError(err)
+
+	//Asset that making a collaborator also creates an account
+	account, uAccountErr := storage.UserAccountGetByUsername(suite.testConfigs.Store, collaboratorInput.UserName)
+	suite.NoError(uAccountErr)
+	suite.NotNil(account)
+
+	planBasics, err := suite.testConfigs.Store.PlanBasicsGetByModelPlanID(plan.ID)
+	suite.NoError(err)
+	suite.NotNil(planBasics)
+
+	oneHourAgo := time.Now().Add(-time.Hour)
+	planBasics.CompleteICIP = &oneHourAgo
+	_, err = suite.testConfigs.Store.PlanBasicsUpdate(suite.testConfigs.Logger, planBasics)
+	suite.NotNil(err)
+
+	testTemplate, expectedSubject, expectedBody := createTemplateCacheHelper(planName, plan)
+	mockEmailTemplateService.
+		EXPECT().
+		GetEmailTemplate(gomock.Eq(email.ModelPlanSuggestedPhaseTemplateName)).
+		Return(testTemplate, nil).
+		AnyTimes()
+
+	mockEmailService.
+		EXPECT().
+		Send(
+			gomock.Any(),
+			gomock.Eq([]string{expectedEmail}),
+			gomock.Any(),
+			gomock.Eq(expectedSubject),
+			gomock.Any(),
+			gomock.Eq(expectedBody),
+		).
+		AnyTimes()
```

Comment:
Should this instead assert a specific number of times we expect an email to be sent. Any asserts it is called 0 or more times, so I don't think it's providing the test we want here.